src/Controls/src/Core/Platform/Android/Extensions/ToolbarExtensions.cs

```diff
@@ -232,9 +232,23 @@ public static void UpdateMenuItems(this AToolbar toolbar,
 			if (sortedToolbarItems == null || previousMenuItems == null)
 				return;
 
-			var context = mauiContext.Context;
 			var menu = toolbar.Menu;
 
+			// menu items can be deleted by Android after switching activities, removing outdated menu items first
+			if (menu != null)
+			{
+				var clonedPreviousMenuItems = previousMenuItems.ToArray();
```

Comment:
Can reduce the memory allocation and copying overhead of `ToArray()` iterating backwards through the collection.

`for (int i = previousMenuItems.Count - 1; i >= 0; i--)`

Reply:
Hi @jsuarezruiz, thank you for the suggestions, updated PR.
I have never added tests like this and creating test environment will take forever, not sure I can do that.